core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepImp.java

```diff
@@ -0,0 +1,173 @@
+package ca.jrvs.apps.grep;
+
+import com.sun.org.slf4j.internal.Logger;
+import com.sun.org.slf4j.internal.LoggerFactory;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+public class JavaGrepImp implements JavaGrep {
+  final Logger logger = LoggerFactory.getLogger(JavaGrep.class);
+  private String regex;
+  private String rootPath;
+  private String outFile;
+
+  public static void main(String[] args) {
+    if (args.length != 3 ) {
+      throw new IllegalArgumentException("USAGE: Java regex rootPath outFile");
+    }
+    JavaGrepImp JavaGrepImp = new JavaGrepImp();
+    JavaGrepImp.setRegex(args[0]);
+    JavaGrepImp.setRootPath(args[1]);
+    JavaGrepImp.setOutFile(args[2]);
+
+    try {
+      JavaGrepImp.process();
+    }catch (Exception ex){
+      JavaGrepImp.logger.error(ex.getMessage());
+    }
+  }
+  /**
+   * Top level search workflow
+   *
+   * @throws IOException
+   */
+  @Override
+  public void process() throws IOException {
+    // initialize a List of string to store all the matched lines
+    List<String> matchedLines = new ArrayList<>();
+    try {
+      //initialize and store all the files
+      for (File file : listFiles(getRootPath())) {
+        // for each file go through parse the file and read each line
```

Comment:
It's nice to write comments. However, if you think another developer can understand you code, then you don't have to put a comment. please remove all inline comments from this method.